daemon/endpoint.go

```diff
@@ -258,6 +258,7 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 	// Now that we have ep.ID we can pin the map from this point. This
 	// also has to happen before the first build took place.
 	if err = ep.PinDatapathMap(); err != nil {
```

Comment:
While you are here, could you change the `PinDatapathMap` method name to `PinDatapathMapLocked`?